### PR TITLE
resolve race condition in maxGauge

### PR DESCRIPTION
**Pull request by dmitsh**
Signed-off-by: Dmitry Shmulevich <dmitry.shmulevich@sysdig.com>

### CODE REVIEW

**Comment by csmarchbanks** [comment]
Thanks for the cc. Reminder to myself to remove this from https://github.com/prometheus/prometheus/pull/5614 if this PR is merged first.

**Comment by dmitsh** [comment]
Could this one be merged? cc @brian-brazil

File: storage/remote/max_gauge.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 )
 
 type maxGauge struct {
-	mtx   sync.Mutex
```
Comment (brian-brazil) [inline storage/remote/max_gauge.go]: A normal mutex is fine. RWMutexes are only suitable when there's many many more reads than writes.

**Comment by dmitsh** [comment]
@brian-brazil addressed

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**